src/Runner.Worker/ActionManager.cs

```diff
@@ -682,12 +682,16 @@ private async Task BuildActionContainerAsync(IExecutionContext executionContext,
                         // * Repo or tag doesn't exist, or isn't public
                         // * Policy validation failed
                         if (ex is WebApi.UnresolvableActionDownloadInfoException)
-                        {
+                        {   // Log the error and fail the JobExtension Initialization.
+                            Trace.Error($"Caught exception from JobExtenion Initialization: {ex}");
+                            executionContext.InfrastructureError(ex.Message);
                             throw;
                         }
                         else
                         {
                             // This exception will be traced as an infrastructure failure
+                            Trace.Error($"Caught exception from JobExtenion Initialization: {ex}");
```

Comment:
We also catch this and throw [here](https://github.com/actions/runner/blob/9958fc0374820841ec0c131fa1259a442593fb1b/src/Runner.Worker/JobExtension.cs#L434). This will result in us double printing in that scenario right?



We could
- Delete the lines in that file and handle it centrally here
- Add this trace.error and infrastructure inside a try catch [here](https://github.com/actions/runner/blob/9958fc0374820841ec0c131fa1259a442593fb1b/src/Runner.Worker/ActionRunner.cs#L92)